addons/udes_stock/models/stock_move_line.py

```diff
@@ -962,22 +1012,76 @@ def _drop_off_criterion_summary(self):
             * Total quantity in move lines
             * Speed of the product (if it is set)
         """
-        self.mapped('product_id')
+        self.mapped("product_id")
         summary = ""
         for product, prod_mls in self.groupby(lambda ml: ml.product_id):
             product_speed = ""
             if product.u_speed_category_id:
                 product_speed = " (speed: {})".format(product.u_speed_category_id.name)
-            summary += "<br>{} x {}{}".format(product.display_name,
-                                              int(sum(prod_mls.mapped('qty_done'))),
-                                              product_speed)
+            summary += "<br>{} x {}{}".format(
+                product.display_name, int(sum(prod_mls.mapped("qty_done"))), product_speed
+            )
         return summary
 
-    @api.constrains('product_id', 'location_id', 'location_dest_id',
-                    'package_id', 'result_package_id', 'lot_id', 'lot_name',
-                    'qty_done')
+    @api.constrains(
+        "product_id",
+        "location_id",
+        "location_dest_id",
+        "package_id",
+        "result_package_id",
+        "lot_id",
+        "lot_name",
+        "qty_done",
+    )
     def _prevent_write_after_done(self):
-        done_lines = self.filtered(lambda x: x.state == 'done')
+        done_lines = self.filtered(lambda x: x.state == "done")
         if done_lines:
-            raise ValidationError(
-                _("Cannot update move lines that are already 'done'."))
+            raise ValidationError(_("Cannot update move lines that are already 'done'."))
+
+    @api.constrains("result_package_id", "u_result_parent_package_id")
+    @api.onchange("result_package_id", "u_result_parent_package_id")
+    def _check_resultant_package_level(self):
+        MoveLine = self.env["stock.move.line"]
+        # Collect move lines with packages related to those being checked which are in progress
+        package_ids = self.mapped("result_package_id") | self.mapped("u_result_parent_package_id")
+        package_mls = MoveLine.search(
+            [
+                ("state", "=", "assigned"),
+                "|",
+                ("result_package_id", "in", package_ids.mapped("id")),
+                ("u_result_parent_package_id", "in", package_ids.mapped("id")),
+            ]
+        )
+
+        for ml in self:
+            storage_format = ml.picking_id.picking_type_id.u_target_storage_format
```

Comment:
You can use: `ml.u_picking_type_id` instead of having to get it from the pick.

Reply:
I'll change that! I copied this from somewhere else 🤔 I'll find that and make a note to sort that